src/utils/viewer/viewer.cpp

```diff
@@ -1302,6 +1382,70 @@ void Viewer::keyPressEvent(KeyEvent& event) {
     case KeyEvent::Key::V:
       invertGravity();
       break;
+#ifdef ESP_BUILD_WITH_VHACD
```

Comment:
How about helper functions in viewer.cpp to encapsulate this logic? Just to keep viewer.cpp more readable.